app/javascript/components/patient/Patient.js

```diff
@@ -68,665 +91,622 @@ class Patient extends React.Component {
+              </div>
+              <div>
+                <b>Preferred Reporting Method:</b>{' '}
+                {(!this.props.details.blocked_sms || !this.props.details.preferred_contact_method?.includes('SMS')) && (
+                  <span>{this.props.details.preferred_contact_method || '--'}</span>
+                )}
+                {this.props.details.blocked_sms && this.props.details.preferred_contact_method?.includes('SMS') && (
+                  <span className="font-weight-bold text-danger">
+                    {this.props.details.preferred_contact_method || '--'}
+                    <Form.Label className="tooltip-whitespace">
+                      <InfoTooltip tooltipTextKey="blockedSMSContactMethod" location="top"></InfoTooltip>
+                    </Form.Label>
+                  </span>
+                )}
+              </div>
+            </div>
+          </Col>
         </Row>
-        <Collapse in={!this.props.hideBody}>
-          <Card.Body className="mx-0 px-0 my-0 py-0">
-            <Row className="g-border-bottom-2 mx-2 pb-4 mb-2"></Row>
-            <Row className="g-border-bottom-2 pb-4 mb-2 mt-4 mx-1">
-              <Col id="address" md="7">
-                <Row>
-                  <Col>
-                    <div className="float-left">
-                      <div className="h4">
-                        <b>ADDRESS</b>
-                      </div>
-                    </div>
-                    <div>
-                      {this.props.goto && (
-                        <Button variant="link" className="pt-0" onClick={() => this.props.goto(1)} aria-label="Edit Button">
-                          <div className="h5">(Edit)</div>
-                        </Button>
-                      )}
-                    </div>
-                    <div className="clearfix"></div>
-                  </Col>
-                </Row>
-                <Row>
-                  <Col className="text-truncate">
-                    {!showDomesticAddress && !showForeignAddress && (
-                      <Row className="py-1">
-                        <Col className="text-truncate">
-                          <span className="none-text">None</span>
-                        </Col>
-                      </Row>
-                    )}
-                    {(showDomesticAddress || showForeignAddress) && (
-                      <Row className="py-1">
-                        <Col>
-                          <b>HOME ADDRESS</b>
-                        </Col>
-                      </Row>
+        {!this.props.editMode && (
+          <div className="details-expander">
+            <Button
+              id="details-expander-link"
+              variant="link"
+              className="p-0"
+              aria-label="Edit Notes"
+              onClick={() => this.setState({ expanded: !this.state.expanded })}>
+              {this.state.expanded ? <FontAwesomeIcon icon={faChevronUp} /> : <FontAwesomeIcon icon={faChevronDown} />}
```

Comment:
Again, not a critique, but my other project has two rotating chevron classes to animate this. LMK if you want me to add it. Does make it look smoother than having the two icons flip-flop. 

👍 for the smooth accordion transition below this though!

Reply:
feel free!

Reply:
Pushed up two commits. One to implement this behavior. One to fix the tests it broke.